### ISSUE TITLE
Generating masks

### ISSUE THREAD

**Issue by sucessfullys** [RollingPlain/SAGE_IVIF#4]
2025-04-24 14:31:10.198222: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2025-04-24 14:31:10.223937: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:485] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2025-04-24 14:31:10.253637: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:8454] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2025-04-24 14:31:10.262509: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1452] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2025-04-24 14:31:10.283713: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2025-04-24 14:31:11.665948: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
Encoder is not available.1: 

Please compile MultiScaleDeformableAttention CUDA op with the following commands:
	`cd mask2former/modeling/pixel_decoder/ops`
	`sh make.sh`

2025-04-24 14:31:18,053 args = Namespace(batch_size=4, epochs=100, crop_size=(192, 256), root_dir='/home2/duanyf/SAGE_IVIF-main/data/FMB/train', use_mask_num=4, cache_mask_num=20, checkpoint_teacher='', checkpoint_student='', learning_rate_Teacher=0.0005, learning_rate_Student=0.002, learning_rate_min=1e-05, report_freq=1, test_freq=10, save='T_EXP-EXP-20250424-143118', seed=2, ini_epoch=1, student_epoch=1)
INFO:TeacherLogger:args = Namespace(batch_size=4, epochs=100, crop_size=(192, 256), root_dir='/home2/duanyf/SAGE_IVIF-main/data/FMB/train', use_mask_num=4, cache_mask_num=20, checkpoint_teacher='', checkpoint_student='', learning_rate_Teacher=0.0005, learning_rate_Student=0.002, learning_rate_min=1e-05, report_freq=1, test_freq=10, save='T_EXP-EXP-20250424-143118', seed=2, ini_epoch=1, student_epoch=1)
2025-04-24 14:31:18,054 args = Namespace(batch_size=4, epochs=100, crop_size=(192, 256), root_dir='/home2/duanyf/SAGE_IVIF-main/data/FMB/train', use_mask_num=4, cache_mask_num=20, checkpoint_teacher='', checkpoint_student='', learning_rate_Teacher=0.0005, learning_rate_Student=0.002, learning_rate_min=1e-05, report_freq=1, test_freq=10, save='T_EXP-EXP-20250424-143118', seed=2, ini_epoch=1, student_epoch=1)
INFO:StudentLogger:args = Namespace(batch_size=4, epochs=100, crop_size=(192, 256), root_dir='/home2/duanyf/SAGE_IVIF-main/data/FMB/train', use_mask_num=4, cache_mask_num=20, checkpoint_teacher='', checkpoint_student='', learning_rate_Teacher=0.0005, learning_rate_Student=0.002, learning_rate_min=1e-05, report_freq=1, test_freq=10, save='T_EXP-EXP-20250424-143118', seed=2, ini_epoch=1, student_epoch=1)
/home2/duanyf/SAGE_IVIF-main/segment_anything/build_sam.py:105: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  state_dict = torch.load(f)
Initializing SAM model and generating 20 masks per image...

Generating masks:   0%|          | 0/1220 [00:00<?, ?it/s]
Why is the mask generation so slow? Can it be accelerated?

**Comment by zhendebufapyq** [comment]
Thank you for your attention to our work. The mask generation process is indeed that slow - we also spent several days generating them. One of the key contributions of our work is precisely to eliminate the dependency on SAM.

**Comment by copawloroous** [comment]
作者您好，请问可以直接公开FMB的Mask_cache吗？

**Comment by zhendebufapyq** [comment]
> 作者您好，请问可以直接公开FMB的Mask_cache吗？

你好!由于mask_cache文件较大,不便直接上传到仓库。建议您直接运行train.py,程序会自动生成所需的mask_cache文件。
